bundles/org.openhab.binding.somfycul/src/main/java/org/openhab/binding/somfycul/internal/CULHandler.java

```diff
@@ -0,0 +1,238 @@
+        // the bridge does not have any channels
+    }
+
+    /**
+     * Executes the given {@link SomfyCommand} for the given {@link Thing} (RTS Device).
+     *
+     * @param somfyDevice the RTS Device which is the receiver of the command.
+     * @param somfyCommand the command to execute
+     * @param rollingCode the current rolling code for the device
+     * @param address the device address
+     * @return true if the command was successfully transmitted to the CUL device, false otherwise
+     */
+    public boolean executeCULCommand(Thing somfyDevice, SomfyCommand somfyCommand, String rollingCode, String address) {
+        // culCommand syntax (basically the serial data payload): Ys + EncryptionKey=A1 + Command + 0 + RollingCode +
+        // Address
+        String culCommand = "YsA1" + somfyCommand.getActionKey() + "0" + rollingCode + address;
+        logger.debug("Send message {} for thing {}", culCommand, somfyDevice.getLabel());
+        return writeString(culCommand);
+    }
+
+    /**
+     * Sends a string to the serial port of this device.
+     * The writing of the msg is executed synchronized, so it's guaranteed that the device doesn't get
+     * multiple messages concurrently.
+     *
+     * @param msg the string to send
+     * @return true, if the message has been transmitted successfully, otherwise false.
+     */
+    protected synchronized boolean writeString(final String msg) {
+        final SerialPortIdentifier localPortId = portId;
+        final OutputStream localOutputStream = outputStream;
+
+        if (localPortId == null || localOutputStream == null) {
+            logger.warn("Cannot write to serial port - port or stream not initialized");
+            return false;
+        }
+
+        logger.debug("Trying to write '{}' to serial port {}", msg, localPortId.getName());
+
+        final long earliestNextExecution = lastCommandTime + COMMAND_DELAY_MS;
+        while (earliestNextExecution > System.currentTimeMillis()) {
+            try {
+                Thread.sleep(COMMAND_DELAY_MS);
+            } catch (InterruptedException ex) {
+                Thread.currentThread().interrupt();
+                return false;
+            }
+        }
+        try {
+            localOutputStream.write((msg + "\n").getBytes());
+            localOutputStream.flush();
+            lastCommandTime = System.currentTimeMillis();
+            return true;
+        } catch (IOException e) {
+            logger.warn("Error writing '{}' to serial port {}: {}", msg, localPortId.getName(), e.getMessage());
+        }
+        return false;
+    }
+
+    @Override
+    public void initialize() {
+        CULConfiguration config = getConfigAs(CULConfiguration.class);
+        if (!validConfiguration(config)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.config-error-missing");
+        } else {
```

Comment:
With an early return you can prevent the nesting and make it more clear.
```suggestion
            return;
        }
```